src/topics/inputs.md

```diff
@@ -143,13 +143,167 @@ The `baseCommand` field will always appear in the final command line before the
+secondary files specified for inputs are required while secondary files specified for outputs are optional. 
+To use `secondaryFiles`, make your primary file 
+(the script you will actually run) a default input and list the other dependencies as `secondaryFiles`.
+An implementation must include matching Files and Directories in the `secondaryFiles` property of the primary file.
+The following example demonstrates the `secondaryFiles` input parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    secondaryFiles: [.idx]
+```
+
+The fields within secondary files are `pattern` and `required`.
+Secondary files are specified using the following micro-DSL for secondary files:
+- If the value is a string, it is transformed to an object with two fields `pattern` and `required`.
+- By default, the value of `required` is null (this indicates default behavior, which may be based on the context).
+- If the value ends with a question mark "?", the question mark is stripped off and the value of the field `required` is set to False.
+- The remaining value is assigned to the field `pattern`.
+
+`pattern` in this context refers how we can reference secondary files,
+with respect to the value, ie if it is an expression, or a string.
+If the value is an expression, the value of self in the expression must be
+the primary input or output File object to which this binding applies.
+The basename, nameroot and nameext fields must be present in self. 
+For CommandLineTool outputs, the path field must also be present. 
+The expression must return a filename string relative to the path to the primary File, 
+a File or Directory object with either path or location and basename fields set, 
+or an array consisting of strings or File or Directory objects. 
+It is legal to reference an unchanged File or Directory object taken from input as a `secondaryFile`. 
+The expression may return "null" in which case there is no `secondaryFile` from that expression.
+
+To work on non-filename-preserving storage systems, 
+portable tool descriptions should avoid constructing new values from location, 
+but should construct relative references using basename or nameroot instead.
+
+If a value in `secondaryFiles` is a string that is not an expression, 
+it specifies that the following pattern should be applied to the path of the primary file 
+to yield a filename relative to the primary File:
+
+- If string begins with one or more caret "^" characters, for each caret, remove the last file extension from the path
+(the last period . and all following characters).
+
+```{code-block} cwl
+inputs:
+  example_file:
+    type: File
+    secondaryFiles: [^.idx]
+```
+
+this will give "example_file.idx".
+
+- If string ends with "?" character, remove the last "?" and mark the resulting secondary file as optional.
+- If there are no file extensions, the path is unchanged.
+Append the remainder of the string to the end of the file path.
```

Comment:
I don't think this is necessary ---> the example with the schema.

Reply:
Below is a direct copy out of specifications -- I don't htink this is necessary but if it is , please reword and make user friendly